app/src/main/java/org/simple/clinic/drugs/selection/DrugListItem.kt

```diff
@@ -45,9 +45,8 @@ data class ProtocolDrugListItem(
     viewBinding.protocoldrugItemName.isChecked = prescribedDrug != null
 
     viewBinding.protocoldrugItemDosageAndFrequency.visibleOrGone(prescribedDrug != null)
-    val medicineFrequency = if (prescribedDrug?.frequency == null) "" else medicineFrequencyToLabelMap[prescribedDrug.frequency]!!.label
-    val dosage = if(prescribedDrug?.dosage == null) "" else prescribedDrug.dosage
-    val dosageAndFrequency = "$dosage, $medicineFrequency"
+    val medicineFrequency = if (prescribedDrug?.frequency == null) null else medicineFrequencyToLabelMap[prescribedDrug.frequency]!!.label
+    val dosageAndFrequency = listOfNotNull(prescribedDrug?.dosage, medicineFrequency).joinToString()
```

Comment:
Honestly, for a couple of items, it's just fine to hardcode. Was IDE showing some lint warning?

Reply:
Here we are ending up creating a new list and then getting string from that. Instead, we can directly just create a string. wdyt?

Reply:
So the benefit of using `joinToString` with `listOfNull` is that it will not show nulls and remove the commas if there is a null string. If we hardcode it we had to handle all the null cases and add empty strings if there was null and also add/remove commas depending on if it had a value or not. So `joinToString` seemed like a better and a more cleaner approach to handle this.

Reply:
Hmm, that makes sense. But readability wise I don't think this is good, for example, we can create a function that creates the dosage and frequency string and do the string manipulations there.

I feel like this would be more understandable than the current implementation, WDYT? Also, this way we don't have to create a new list of objects and then covert them to a string.

```kt
private fun dosageAndFrequency(dosage: String?, frequency: MedicineFrequency): String {
  return if (dosage != null) {
    "$dosage, $frequency"
  } else {
    "$frequency"
  }
}
```

Reply:
We will have to add a check for frequency as well `val medicineFrequency = if (prescribedDrug?.frequency == null) null else medicineFrequencyToLabelMap[prescribedDrug.frequency]!!.label` because we don't want to display `None` frequencies. 
It just seems like we are doing a lot of things here to do something that those methods handle already. 😅

Reply:
Cool, sounds good 👍🏾